tests/settings.py

```diff
@@ -94,8 +94,11 @@ def canonicalize(self) -> None:
         self._canonicalized = True
 
     @property
-    def userinfos(self) -> List[UserInfo]:
-        return [UserInfo.parse_obj(x) for x in self.test_objects["user_info"]]
+    def userinfos(self) -> dict[str, UserInfo]:
+        return {
+            t: UserInfo.parse_obj(d)
+            for t, d in self.test_objects["user_info"].items()
+        }
```

Comment:
Hah.  The one time I actually don't just dive for a dict, it would have been the right choice.